src/redux-middleware/pullBeforeMove.ts

```diff
@@ -0,0 +1,35 @@
+import { ThunkMiddleware } from 'redux-thunk'
+import pull from '../action-creators/pull'
+import { isPending, pathExists, simplifyPath } from '../selectors'
+import { hashContext, head, parentOf, pathToContext } from '../util'
+import { State } from '../util/initialState'
+
+/**
+ * Middleware that fetches any pending descendants before a thought-move operation.
+ */
+const pullBeforeMove: ThunkMiddleware<State> = ({ getState, dispatch }) => {
```

Comment:
Does this replace `flushMoves`? Or is that still needed?

https://github.com/cybersemics/em/blob/18a3dc9b98386e4225b7e30dd5a8577ceeb0721e/src/redux-middleware/pushQueue.ts#L107-L133

Reply:
`pendingMoves` is being used for pulling the [source thoughts](https://github.com/cybersemics/em/blob/a44a411697efb8112440386c3f4cb7f343bca746/src/reducers/existingThoughtMove.ts#L359) that are not yet loaded, so we can't replace that yet

Reply:
It would take some refactoring, but is it possible to handle source and destination pulling using the same mechanism? That would be much cleaner than having two completely separate mechanisms.